specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-07-01-preview/dataConnectorDefinitions.json

```diff
@@ -0,0 +1,758 @@
+      "allOf": [
+        {
+          "$ref": "../../../common/2.0/types.json#/definitions/ResourceWithEtag"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "An Azure resource, which encapsulate the entire info requires to display a data connector page in Azure portal,\r\nand the info required to define data connections.",
+          "type": "object",
+          "$ref": "#/definitions/ConnectorDefinitionBase",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ConnectorDefinitionBaseArmObjectWrapperWithConverter": {
+      "required": [
+        "properties"
+      ],
+      "type": "object",
+      "description": "Encapsulate the data connector definition object",
+      "allOf": [
+        {
+          "$ref": "../../../common/2.0/types.json#/definitions/ResourceWithEtag"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Gets or sets the nested level of properties which contains the resource content",
+          "type": "object",
+          "$ref": "#/definitions/ConnectorDefinitionBase",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "CustomizableConnectorDefinition": {
+      "description": "Connector definition for ConnectorDefinitionKind 'Customizable'.",
+      "required": [
+        "connectorUiConfig"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ConnectorDefinitionBase"
+        }
+      ],
+      "properties": {
+        "connectorUiConfig": {
+          "$ref": "#/definitions/CustomizableConnectorUiConfig"
+        },
+        "connectionsConfig": {
+          "$ref": "#/definitions/CustomizableConnectionsConfig"
+        }
+      }
+    },
+    "CustomizableConnectionsConfig": {
+      "description": "The UiConfig for 'Customizable' connector definition kind.",
+      "required": [
+        "templateSpecVersion",
+        "templateSpecName"
+      ],
+      "type": "object",
+      "properties": {
+        "templateSpecName": {
+          "description": "Gets or sets the template name. The template includes ARM templates that can be created by the connector, usually it will be the dataConnectors ARM templates.",
+          "type": "string"
+        },
+        "templateSpecVersion": {
+          "description": "Gets or sets the template version.",
+          "type": "string"
+        }
+      }
+    },
+    "CustomizableConnectorUiConfig": {
+      "description": "The UiConfig for 'Customizable' connector definition kind.",
+      "required": [
+        "dataTypes",
+        "descriptionMarkdown",
+        "graphQueries",
+        "instructionSteps",
+        "permissions",
+        "publisher",
+        "sampleQueries",
+        "title",
+        "connectivityCriteria"
+      ],
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Gets or sets custom connector id. optional field.",
```

Comment:
Is it an azure resource ID? If so, please consider add `format: "arm-id"`

refs: https://github.com/Azure/autorest/blob/800f47e65e201e0abf64cb24362f1d66357e0930/docs/extensions/readme.md#x-ms-arm-id-details

Reply:
it is not arm id. 
it is  internal id, to give the ability to override some hard code definition to a custom definition by the user